### PR TITLE
Only log error when tag sync fails

### LINKED ISSUE #28843: unable to PushUpdateAddTag to Repo: unable to get tag Commit: object does not exist
### Description

image

Using gogit and migrate https://github.com/torvalds/linux with releases

https://github.com/torvalds/linux/tree/v2.6.11
image

not sure whether this is a bug

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build

### Database

None

### PR DESCRIPTION
**Pull request by lunny**
Fix #28843

This PR will bypass the pushUpdateTag to database failure when syncAllTags. An error log will be recorded.

### CODE REVIEW

File: modules/repository/repo.go [hunk 1]
```diff
@@ -352,7 +352,11 @@ func SyncReleasesWithTags(ctx context.Context, repo *repo_model.Repository, gitR
 		}
 
 		if err := PushUpdateAddTag(ctx, repo, gitRepo, tagName, sha1, refname); err != nil {
-			return fmt.Errorf("unable to PushUpdateAddTag: %q to Repo[%d:%s/%s]: %w", tagName, repo.ID, repo.OwnerName, repo.Name, err)
+			// ref https://github.com/torvalds/linux/tree/v2.6.11
+			// this is a tree object, not a tag object which created before git
+			if sha1 != "c39ae07f393806ccf406ef966e9a15afc43cc36a" {
+				return fmt.Errorf("unable to PushUpdateAddTag: %q to Repo[%d:%s/%s]: %w", tagName, repo.ID, repo.OwnerName, repo.Name, err)
+			}
```
Comment (wxiaoguang) [inline modules/repository/repo.go]: Does it really make sense to return "err" here?

Why not just log the error and continue the "WalkReferences" to tolerate any corrupted object.

File: modules/repository/repo.go [hunk 2]
```diff
@@ -352,7 +352,11 @@ func SyncReleasesWithTags(ctx context.Context, repo *repo_model.Repository, gitR
 		}
 
 		if err := PushUpdateAddTag(ctx, repo, gitRepo, tagName, sha1, refname); err != nil {
-			return fmt.Errorf("unable to PushUpdateAddTag: %q to Repo[%d:%s/%s]: %w", tagName, repo.ID, repo.OwnerName, repo.Name, err)
+			// ref https://github.com/torvalds/linux/tree/v2.6.11
```
Comment (silverwind) [inline modules/repository/repo.go]:
```suggestion
			// ref https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tag/?h=v2.6.11
```

your link is 404, this one works

**Comment by silverwind** [comment]
I wonder if this is really the only place to ignore this hash or if it needs to be lower level. Does `git` also have special handling for this hash?

File: modules/repository/repo.go (hunk 2 above)
Comment (wxiaoguang) [reply modules/repository/repo.go]: I would prefer to do this: https://github.com/go-gitea/gitea/pull/29295#pullrequestreview-1893511989 🤣

Comment (silverwind) [reply modules/repository/repo.go]: I would prefer a solution without a hardcoded hash, but if it has to stay, the comment link needs to work at least 😆.

File: modules/repository/repo.go [hunk 3]
```diff
@@ -352,7 +352,9 @@ func SyncReleasesWithTags(ctx context.Context, repo *repo_model.Repository, gitR
 		}
 
 		if err := PushUpdateAddTag(ctx, repo, gitRepo, tagName, sha1, refname); err != nil {
-			return fmt.Errorf("unable to PushUpdateAddTag: %q to Repo[%d:%s/%s]: %w", tagName, repo.ID, repo.OwnerName, repo.Name, err)
+			// sometimes, some tags will be sync failed. i.e. https://github.com/torvalds/linux/tree/v2.6.11
```
Comment (wxiaoguang) [inline modules/repository/repo.go]:
```suggestion
			// sometimes, some tags will be sync failed. i.e. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tag/?h=v2.6.11
```

Thanks to silverwind

**Review by wxiaoguang — APPROVED** [review]

File: modules/repository/repo.go [hunk 4]
```diff
@@ -352,7 +352,9 @@ func SyncReleasesWithTags(ctx context.Context, repo *repo_model.Repository, gitR
 		}
 
 		if err := PushUpdateAddTag(ctx, repo, gitRepo, tagName, sha1, refname); err != nil {
-			return fmt.Errorf("unable to PushUpdateAddTag: %q to Repo[%d:%s/%s]: %w", tagName, repo.ID, repo.OwnerName, repo.Name, err)
+			// sometimes, some tags will be sync failed. i.e. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tag/?h=v2.6.11
+			// this is a tree object, not a tag object which created before git
+			log.Error("unable to PushUpdateAddTag: %q to Repo[%d:%s/%s]: %w", tagName, repo.ID, repo.OwnerName, repo.Name, err)
```
Comment (wxiaoguang) [inline modules/repository/repo.go]:
```suggestion
			log.Error("unable to PushUpdateAddTag: %q to Repo[%d:%s/%s]: %v", tagName, repo.ID, repo.OwnerName, repo.Name, err)
```

**Review by delvh — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Edited in a more accurate title.

**Pull request merged**